### PR TITLE
refactored existing test cases, added new test cases into smart table…

### PR DESCRIPTION
**Pull request by fahimarshaddevsinc**
… page

### Please read and mark the following check list before creating a pull request (check one with "x"):

 - [ ] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/ngx-admin/blob/master/CONTRIBUTING.md) guide.
 - [ ] I read and followed the [New Feature Checklist](https://github.com/akveo/ngx-admin/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:

### CODE REVIEW

File: cypress/integration/smartTablePageTestCases.spec.js [hunk 1]
```diff
@@ -1,16 +1,80 @@
...
 
     beforeEach('check smart table is opening', () => {
         cy.visit('/')
         navigateTo.smartTablePage()
     })
 
-    it('should check table edit functionality', () => {
-        onSmartTablePage.check_table_edit_functionality_one('Jacob', '24')
-        //onSmartTablePage.check_table_edit_functionality('Larry', '24')
+    it('Should verify that changes in any field are edited correctly', () => {
```
Comment (hassanatdevsinc) [inline cypress/integration/smartTablePageTestCases.spec.js]: It block should always start with the word 'verify if'

File: cypress/support/page_object/smartTablePage.js [hunk 2]
```diff
@@ -1,42 +1,124 @@
 /// <reference types="cypress" />
 
+// import { element } from "protractor"
+
 export class smartTablePage {
 
     //ALL LOCATORS
     elements = {
-        table_row_edit_icon:(personName) =>  cy.get('table').contains('tr', personName).then ( tableRow => {
+        first_row_edit_icon:(row) => cy.get('tbody').find('tr').first().find('.nb-edit'),
```
Comment (hassanatdevsinc) [inline cypress/support/page_object/smartTablePage.js:9]: from now onward try using data-test-id as locators. I will discuss it with you in-person

File: cypress/integration/smartTablePageTestCases.spec.js (hunk 1 above)
Comment (fahimarshaddevsinc) [reply cypress/integration/smartTablePageTestCases.spec.js]: resolved.

File: cypress/support/page_object/smartTablePage.js (hunk 2 above)
Comment (fahimarshaddevsinc) [reply cypress/support/page_object/smartTablePage.js:9]: noted.

**Pull request merged**